src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/common.hpp

```diff
@@ -103,35 +124,49 @@ static constexpr size_t vec_len_f32_avx2 = vec_len_avx2 / sizeof(float);
         };
         return _mm256_loadu_si256(&mask[N7]);
     }
-    inline __m256 mm256_uni_loadu_ps(const float* a) {
-        return _mm256_loadu_ps(a);
-    }
-    inline void mm256_uni_storeu_ps(float* a,  __m256 v) {
-        _mm256_storeu_ps(a, v);
+
+    inline __m256 mm256_uni_loadu_ps(const float* addr) {
+        return _mm256_loadu_ps(addr);
     }
 
-    inline __m256 mm256_uni_loadu_ps(const ov::bfloat16* a) {
-        auto vec_bf16 = _mm_loadu_si128(reinterpret_cast<const __m128i*>(a));
+    inline __m256 mm256_uni_loadu_ps(const ov::bfloat16* addr) {
+        auto vec_bf16 = _mm_loadu_si128(reinterpret_cast<const __m128i*>(addr));
         auto o = _mm256_castsi256_ps(_mm256_slli_epi32(_mm256_cvtepu16_epi32(vec_bf16), 16));
         return o;
     }
 
-    inline __m256 mm256_uni_loadu_tail_ps(const float* a, const size_t count) {
+    inline __m256 mm256_uni_loadu_ps(const ov::float16* addr) {
+        auto vec_f16 = _mm_loadu_si128(reinterpret_cast<const __m128i*>(addr));
+        auto o = _mm256_cvtph_ps(vec_f16);
+        return o;
+    }
+
+    inline __m256 mm256_uni_loadu_tail_ps(const float* addr, const size_t count) {
         auto mask = get_mask(count);
-        return _mm256_maskload_ps(a, mask);
+        return _mm256_maskload_ps(addr, mask);
     }
 
-    inline __m256 mm256_uni_loadu_tail_ps(const ov::bfloat16* a, const size_t count) {
+    inline __m256 mm256_uni_loadu_tail_ps(const ov::bfloat16* addr, const size_t count) {
         assert("AVX2 version of bfloat16 tail load is just for compilation pass");
         ov::bfloat16 tmp_values[8] = {0};
-        std::memcpy(tmp_values, a, count * sizeof(ov::bfloat16));
+        std::memcpy(tmp_values, addr, count * sizeof(ov::bfloat16));
+        return mm256_uni_loadu_ps(tmp_values);
+    }
+
+    inline __m256 mm256_uni_loadu_tail_ps(const ov::float16* addr, const size_t count) {
```

Comment:
will f16 load_tail be used by AVX2 platform ? If not, please also add an assertion just like bf16

Reply:
It's referenced in [softmax](https://github.com/luo-cheng2021/openvino/blob/4172152a8e69404722c63c4425eb65a294bc2fe4/src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/softmax_kernel.hpp#L274), if attn_mask is f16, the logic will be used.